src/ini_type.h

```diff
@@ -21,12 +23,12 @@ enum IniGroupType {
 
 /** A single "line" in an ini file. */
 struct IniItem {
-	IniItem *next; ///< The next item in this group
-	char *name;    ///< The name of this item
-	char *value;   ///< The value of this item
-	char *comment; ///< The comment associated with this item
+	IniItem *next;                    ///< The next item in this group
+	std::string name;                 ///< The name of this item
+	opt::optional<std::string> value; ///< The value of this item
```

Comment:
I'm not spotting any code in this commit that makes a distinction between not existing and an empty string, so it looks like the optional could be dropped to simplify a number of checks to just "value.empty()".

Reply:
It can't, actually. I'd have to look back to point out the exact line (and in which commit it is), but some settings depend on it.

I.e. `digit_group_separator = ` means something different to `digit_group_separator =  ""`.

Reply:
That example makes sense. Good to know you already tried that option :)